src/components/implementation/no_interface/vkernel/vkernel.c

```diff
@@ -85,6 +85,9 @@ cos_init(void)
 	vk_info.termthd = cos_thd_alloc(vk_cinfo, vk_cinfo->comp_cap, vk_terminate, NULL);
 	assert(vk_info.termthd);
 
+	while (!(cycs = cos_hw_cycles_per_usec(BOOT_CAPTBL_SELF_INITHW_BASE))) ;
```

Comment:
Wow, I should have done it! Yes, I had that in mind, but wasn't sure if you'd approve of cos_kernel_api blocking for that. Thank you, I'll keep that in mind.